src/lib.rs

```diff
@@ -11,6 +11,898 @@
+                        config_path.clone(),
+                        &runtime_wrapper,
+                    )
+                    .await;
+                    trace!(
+                        "{}:{} - After attempt to create new app for client_id: {} app_name: {}",
+                        UP_CLIENT_VSOMEIP_TAG,
+                        UP_CLIENT_VSOMEIP_FN_TAG_APP_EVENT_LOOP,
+                        client_id,
+                        app_name
+                    );
+
+                    if let Err(err) = new_app_res {
+                        error!("Unable to create new app: {:?}", err);
+                        Self::return_oneshot_result(Err(err), return_channel).await;
+                        continue;
+                    }
+
+                    let mut client_id_app_mapping = CLIENT_ID_APP_MAPPING.write().await;
+                    if let std::collections::hash_map::Entry::Vacant(e) =
+                        client_id_app_mapping.entry(client_id)
+                    {
+                        e.insert(app_name.clone());
+                        trace!(
+                            "Inserted client_id: {} and app_name: {} into CLIENT_ID_APP_MAPPING",
+                            client_id,
+                            app_name
+                        );
+                        Self::return_oneshot_result(Ok(()), return_channel).await;
+                    } else {
+                        let err_msg = format!("Already had key. Somehow we already had an application running for client_id: {}", client_id);
+                        let err = UStatus::fail_with_code(UCode::ALREADY_EXISTS, err_msg);
+                        Self::return_oneshot_result(Err(err), return_channel).await;
+                    }
+                }
+            }
+            trace!("Hit bottom of event loop");
+        }
+    }
+
+    async fn register_listener_internal(
+        source_filter: UUri,
+        sink_filter: Option<UUri>,
+        registration_type: RegistrationType,
+        msg_handler: MessageHandlerFnPtr,
+        application_wrapper: &mut UniquePtr<ApplicationWrapper>,
+        _runtime_wrapper: &UniquePtr<RuntimeWrapper>,
+    ) -> Result<(), UStatus> {
+        trace!(
+            "{}:{} - Attempting to register: source_filter: {:?} & sink_filter: {:?}",
+            UP_CLIENT_VSOMEIP_TAG,
+            UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+            source_filter,
+            sink_filter
+        );
+
+        match registration_type {
+            RegistrationType::Publish(_) => {
+                trace!(
+                    "{}:{} - Registering for Publish style messages.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+                let (_, service_id) = split_u32_to_u16(source_filter.ue_id);
+                // let instance_id = vsomeip::ANY_INSTANCE; // TODO: Set this to 1? To ANY_INSTANCE?
```

Comment:
What should this be set to? Wondering if you know @int0x27 / @stevenhartley

Reply:
Need to understand what this is used for. we are not telling SOME/Ip that we want to "register for all published events" but we need to subscribe to only the events/messages that local (high compute) entities want to subscribe to.

Reply:
You can make a helper function that returns `(instance_id, service_id)` from `ue_id` as specified in: https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/someip.adoc#uuris